### ISSUE TITLE
Kernel panic in hi_dvb pes packet handler

### ISSUE THREAD

**Issue by peterurbanec** [beyonwiz-dev/enigma2#18]
I was using timeshift and seeking back and forward. After a few minutes of this, the V2 locked up. The serial console had the following crash, indicating a problem in the driver.

```
Kernel panic - not syncing: Attempt to queue invalid stream id 0x8d                                                                                                                                                     
CPU: 3 PID: 7324 Comm: enigma2 Not tainted 4.4.35 #1                                                                                                                                                                                                                    
Hardware name: bigfish                                                                                                                                                                                                                                             
[<c00215f4>] (unwind_backtrace) from [<c001b8e0>] (show_stack+0x20/0x24)                                                                                                                                                                                                       
[<c001b8e0>] (show_stack) from [<c03c62b8>] (dump_stack+0x90/0xa4)                                                                                                                                                                                                             
[<c03c62b8>] (dump_stack) from [<c00ef628>] (panic+0xac/0x20c)                                                                                                                                                                                                                 
[<c00ef628>] (panic) from [<bf736da8>] (new_pes_packet+0x37c/0x760 [hi_dvb])                                                                                                                                                                                                   
[<bf736da8>] (new_pes_packet [hi_dvb]) from [<bf73745c>] (mpegts_push_data+0x2d0/0x59c [hi_dvb])
[<bf73745c>] (mpegts_push_data [hi_dvb]) from [<bf737eb4>] (handle_ts_packet+0xd8/0x1ac [hi_dvb])
[<bf737eb4>] (handle_ts_packet [hi_dvb]) from [<bf6f0aec>] (hi_dvb_write_to_decoder+0x40/0x48 [hi_dvb])
[<bf6f0aec>] (hi_dvb_write_to_decoder [hi_dvb]) from [<c05bfcb8>] (dvb_dmx_swfilter_packet+0x354/0x598)
[<c05bfcb8>] (dvb_dmx_swfilter_packet) from [<c05c0080>] (dvb_dmx_swfilter+0x100/0x1a0)
[<c05c0080>] (dvb_dmx_swfilter) from [<c05c16d4>] (dvbdmx_write+0x1e4/0x38c)
[<c05c16d4>] (dvbdmx_write) from [<c05bd5f4>] (dvb_dvr_write+0x78/0xb8)
[<c05bd5f4>] (dvb_dvr_write) from [<c013faf4>] (__vfs_write+0x38/0xec)
[<c013faf4>] (__vfs_write) from [<c01403a8>] (vfs_write+0xa0/0x178)
[<c01403a8>] (vfs_write) from [<c0140c4c>] (SyS_write+0x4c/0xa0)
[<c0140c4c>] (SyS_write) from [<c0016940>] (ret_fast_syscall+0x0/0x3c)
CPU1: stopping
CPU: 1 PID: 0 Comm: swapper/1 Not tainted 4.4.35 #1
Hardware name: bigfish
[<c00215f4>] (unwind_backtrace) from [<c001b8e0>] (show_stack+0x20/0x24)
[<c001b8e0>] (show_stack) from [<c03c62b8>] (dump_stack+0x90/0xa4)
[<c03c62b8>] (dump_stack) from [<c001f07c>] (handle_IPI+0x298/0x2d0)
[<c001f07c>] (handle_IPI) from [<c000951c>] (gic_handle_irq+0x98/0x9c)
[<c000951c>] (gic_handle_irq) from [<c001c314>] (__irq_svc+0x54/0x70)
Exception stack(0xdf521f48 to 0xdf521f90)
1f40:                   00000000 dfbc2348 57487052 c002b240 ffffe000 00000001
1f60: c0c30408 c081ec48 c0c2fa19 c0ba2cd4 00000000 df521fa4 df521fa8 df521f98
1f80: c0017b74 c0017b78 600b0013 ffffffff
[<c001c314>] (__irq_svc) from [<c0017b78>] (arch_cpu_idle+0x48/0x4c)
[<c0017b78>] (arch_cpu_idle) from [<c00753d4>] (default_idle_call+0x38/0x3c)
[<c00753d4>] (default_idle_call) from [<c00754fc>] (cpu_startup_entry+0x124/0x2bc)
[<c00754fc>] (cpu_startup_entry) from [<c001eb58>] (secondary_start_kernel+0x168/0x188)
[<c001eb58>] (secondary_start_kernel) from [<000095ac>] (0x95ac)
CPU2: stopping
CPU: 2 PID: 0 Comm: swapper/2 Not tainted 4.4.35 #1
Hardware name: bigfish
[<c00215f4>] (unwind_backtrace) from [<c001b8e0>] (show_stack+0x20/0x24)
[<c001b8e0>] (show_stack) from [<c03c62b8>] (dump_stack+0x90/0xa4)
[<c03c62b8>] (dump_stack) from [<c001f07c>] (handle_IPI+0x298/0x2d0)
[<c001f07c>] (handle_IPI) from [<c000951c>] (gic_handle_irq+0x98/0x9c)
[<c000951c>] (gic_handle_irq) from [<c001c314>] (__irq_svc+0x54/0x70)
Exception stack(0xdf523f48 to 0xdf523f90)
3f40:                   00000000 dfbcf348 30586962 c002b240 ffffe000 00000002
3f60: c0c30408 c081ec48 c0c2fa19 c0ba2cd4 00000000 df523fa4 df523fa8 df523f98
3f80: c0017b74 c0017b78 60010013 ffffffff
[<c001c314>] (__irq_svc) from [<c0017b78>] (arch_cpu_idle+0x48/0x4c)
[<c0017b78>] (arch_cpu_idle) from [<c00753d4>] (default_idle_call+0x38/0x3c)
[<c00753d4>] (default_idle_call) from [<c00754fc>] (cpu_startup_entry+0x124/0x2bc)
[<c00754fc>] (cpu_startup_entry) from [<c001eb58>] (secondary_start_kernel+0x168/0x188)
[<c001eb58>] (secondary_start_kernel) from [<000095ac>] (0x95ac)
CPU0: stopping
CPU: 0 PID: 0 Comm: swapper/0 Not tainted 4.4.35 #1
Hardware name: bigfish
[<c00215f4>] (unwind_backtrace) from [<c001b8e0>] (show_stack+0x20/0x24)
[<c001b8e0>] (show_stack) from [<c03c62b8>] (dump_stack+0x90/0xa4)
[<c03c62b8>] (dump_stack) from [<c001f07c>] (handle_IPI+0x298/0x2d0)
[<c001f07c>] (handle_IPI) from [<c000951c>] (gic_handle_irq+0x98/0x9c)
[<c000951c>] (gic_handle_irq) from [<c001c314>] (__irq_svc+0x54/0x70)
Exception stack(0xc0b91f08 to 0xc0b91f50)
1f00:                   00000000 dfbb5348 4d7a9880 c002b240 ffffe000 c0c4c000
1f20: c0c30408 c081ec48 c0c2fa19 c0ba2cd4 e03fc900 c0b91f64 c0b91f68 c0b91f58
1f40: c0017b74 c0017b78 600f0013 ffffffff
[<c001c314>] (__irq_svc) from [<c0017b78>] (arch_cpu_idle+0x48/0x4c)
[<c0017b78>] (arch_cpu_idle) from [<c00753d4>] (default_idle_call+0x38/0x3c)
[<c00753d4>] (default_idle_call) from [<c00754fc>] (cpu_startup_entry+0x124/0x2bc)
[<c00754fc>] (cpu_startup_entry) from [<c08179cc>] (rest_init+0x88/0x8c)
[<c08179cc>] (rest_init) from [<c0b05d30>] (start_kernel+0x3ac/0x3b8)
---[ end Kernel panic - not syncing: Attempt to queue invalid stream id 0x8d
```

**Comment by atvcaptain** [comment]
fixed with update with drivers 2019.03.20 please check

**Comment by peterurbanec** [comment]
I have not seen this problem again, after updating the drivers.
